CONTRIBUTING.md

```diff
@@ -0,0 +1,900 @@
+
+Do put an extra space after the first brace and before the final brace in single-line functions.
+
+```js
+// Good
+if (condition) doSomething()
+
+if (condition) {
+	doSomething()
+}
+
+if (condition) {
+	doSomething()
+} else {
+	doSomethingElse()
+}
+
+if (condition) {
+	doSomething()
+	doSomethingElse()
+}
+
+setTimeout(function () { doSomething() }, 0)
+
+setTimeout(function () {
+	doSomething()
+}, 0)
+
+// Bad
+if (condition) { doSomething(); doSomethingElse() }
+
+if (condition)
+	doSomething()
+
+setTimeout(function () { doSomething(); doSomethingElse() }, 0)
+
+setTimeout(function () {doSomething()}, 0) // Body needs some space
+```
+
+### Function declarations
+
+Function declarations should always take more than one line.
+
+```js
+// Good
+function foo() {
+	return bar
+}
+
+// Bad
+function foo() { return bar }
+```
+
+### `if`-`else`
+
+Be consistent with braces in `if`-`else` statements.
+
+```js
+// Good
+if (condition) doSomething()
+else doSomethingElse()
+
+if (condition) {
+	doSomething()
+} else {
+	doSomethingElse()
+}
+
+if (condition) {
+	doSomething()
+} else if (anotherCondition) {
+	doSomethingElse()
+} else {
+	doAnotherThing()
+}
+
+// Okay
+if (condition) doSomething()
+else if (anotherCondition) doSomethingElse()
+else doAnotherThing()
+
+// Bad
+if (condition) doSomething()
+else {
+	doSomethingElse()
+}
+
+if (condition) doSomething()
+else if (anotherCondition) {
+	doSomethingElse()
+} else doAnotherThing()
+
+
+if (condition) {
+	doSomething()
+} else doSomethingElse()
+```
+
+### `do`-`while` loops
+
+This should be avoided in most cases, since the only good use case is if the intent is "do this, and then check" each iteration. This should only be used in cases like below:
+
+```js
+// Form 1
+doSomething()
+
+while (condition) {
+	doSomething()
+}
+
+// Form 2
+while (true) {
+	doSomething()
+	if (!condition) break
+}
+```
+
+Always use braces. Also, there must be a semicolon at the end of the `while` condition. This is the only exception to the semicolon rule, since the omission is a syntax error. The semicolon is never automatically inserted before ES6.
+
+```js
+// Good
+do {
+	doSomething()
+} while (condition);
+
+// Bad
+do doSomething()
+while (condition);
+
+// Syntax errors
+do {
+	doSomething()
+} while (condition)
+
+do doSomething()
+while (condition)
+```
+
+### Empty blocks
+
+Mark empty blocks as intentionally empty via a comment or similar. Don't just leave an empty block and/or semicolon.
+
+```js
+// Good
+for (var i = 0; i < list.length && cond(list[i]); i++) {
+	// empty
+}
+
+// Bad
+for (var i = 0; i < list.length && cond(list[i]); i++) {}
+
+// Also bad
+for (var i = 0; i < list.length && cond(list[i]); i++) {
+	;
+}
+
+// Even worse
+for (var i = 0; i < list.length && cond(list[i]); i++);
+```
```

Comment:
`//empty` is not a good comment. It should describe why the code is the way it is.

Also empty block may unintentially invite a contributor to add code where it's not wise to do so.